main.tf

```diff
@@ -307,6 +269,67 @@ resource "ibm_cos_bucket" "cos_bucket1" {
   }
 }
 
+locals {
+  expiration_or_archiving_rule_enabled = (length(local.expire_enabled) != 0 || length(local.archive_enabled) != 0)
+
+  create_cos_bucket  = (var.kms_encryption_enabled && var.create_cos_bucket) ? true : false
+  create_cos_bucket1 = (!var.kms_encryption_enabled && var.create_cos_bucket) ? true : false
+
+  cos_bucket_resource = local.create_cos_bucket ? ibm_cos_bucket.cos_bucket : local.create_cos_bucket1 ? ibm_cos_bucket.cos_bucket1 : null
+
+  ## Only one of these values can be set, leaving 2 of 3 null, compact function removes nulls.
+  ## We then take the only value left in the list
+  cos_region = compact([var.region, var.cross_region_location, var.single_site_location])[0]
+}
+
+resource "time_sleep" "wait_for_cos_bucket_lifecycle" {
```

Comment:
there is a timing issue because after the first apply unless I waited a couple seconds to reapply it would fail saying it couldn't find the lifecycle policy. I will create a provider issue and link in a comment above sleep.

Reply:
````hcl
TestRunAdvancedExample 2024-11-06T23:58:29Z command.go:185: Planning failed. Terraform encountered an error while generating this plan.
TestRunAdvancedExample 2024-11-06T23:58:29Z command.go:185: 
TestRunAdvancedExample 2024-11-06T23:58:29Z command.go:185: ╷
Error: AdvancedExample 2024-11-06T23:58:29Z command.go:185: │ Error: [ERROR] Error getting Lifecycle Configuration for the bucket cos-advanced-91sqap-bucket-3-3hva
TestRunAdvancedExample 2024-11-06T23:58:29Z command.go:185: │ 
TestRunAdvancedExample 2024-11-06T23:58:29Z command.go:185: │   with module.cos_bucket3.ibm_cos_bucket_lifecycle_configuration.cos_bucket_lifecycle[0],
TestRunAdvancedExample 2024-11-06T23:58:29Z command.go:185: │   on ../../main.tf line 285, in resource "ibm_cos_bucket_lifecycle_configuration" "cos_bucket_lifecycle":
TestRunAdvancedExample 2024-11-06T23:58:29Z command.go:185: │  285: resource "ibm_cos_bucket_lifecycle_configuration" "cos_bucket_lifecycle" {
````

The error can be seen here in this [run](https://github.com/terraform-ibm-modules/terraform-ibm-cos/actions/runs/11713814566/job/32627408555#step:7:3885) when I removed the sleep. It doesn't occur every time, but often enough.

Reply:
as per internal discussions, the issue happens on the plan AFTER apply has completed. Therefor it seems a bug with the `ibm_cos_bucket_lifecycle_configuration` resource. If its not fully ready, it should of been marked as complete by terraform. Lets create a provider issue, and hold off fix this migration until its fixed

Reply:
https://github.com/IBM-Cloud/terraform-provider-ibm/issues/5778